indexer/src/utils/filter.ts

```diff
@@ -0,0 +1,62 @@
+// Starknet
+import { Transaction, Event } from "../deps.ts";
+
+const KAKAROT_ADDRESS = Deno.env.get("KAKAROT_ADDRESS");
+if (KAKAROT_ADDRESS === undefined) {
+  throw new Error("ENV: KAKAROT_ADDRESS is not set");
+}
+
+export const isKakarotTransaction = (transaction: Transaction) => {
+  // Filter out transactions that are not related to Kakarot.
+  // callArrayLen <- calldata[0]
+  // to <- calldata[1]
+  // selector <- calldata[2];
+  // dataOffset <- calldata[3]
+  // dataLength <- calldata[4]
+  // calldataLen <- calldata[5]
+  const calldata = transaction.invokeV1?.calldata;
+  if (calldata === undefined) {
+    console.error("No calldata in transaction");
+    console.error(JSON.stringify(transaction, null, 2));
+    return false;
+  }
+  const to = calldata[1];
+  if (to === undefined) {
+    console.error("No `to` field in calldata of transaction");
+    console.error(JSON.stringify(transaction, null, 2));
+    return false;
+  }
+  // TODO(Greged93): replace this with a more robust check.
+  // ⚠️ The existence of `to` field in invoke calldata in RPC is not enforced by protocol.
+  // Forks or modifications of the kkrt-labs/kakarot-rpc codebase could break this check.
+  if (BigInt(to) !== BigInt(KAKAROT_ADDRESS!)) {
+    console.log("✅ Skipping transaction that is not related to Kakarot");
+    return false;
+  }
+  return true;
+};
+
+export const ethValidationFailed = (event: Event) => {
```

Comment:
As you are using a lot the `data` field of `event` in this method, maybe you could have something like

```ts
  const { data } = event;
```

at the beginning of the function, doing then:

```ts
if (data.length === 0) {
    return false;
  }
```